src/kps/business/RouteCalculator.java

```diff
@@ -9,58 +11,82 @@
 
-public class RouteCalculator {
-    private List<Route> routes;
+public class RouteCalculator implements IRouteCalculator {
+    private final List<Route> routes;
+    private final List<CalculatedRoute> possiblePaths = new ArrayList<>();
+    private final Set<String> visitedNodes = new HashSet<>();
+    private final @NotNull Mail mailDelivery;
 
-    public RouteCalculator(Simulation simulation) {
-        this.routes = simulation.getUniqueRoutes().stream().filter(r -> !r.isDiscontinued()).collect(Collectors.toList());
+    public RouteCalculator(@NotNull Simulation simulation, @NotNull Mail mailDelivery) {
+        this.mailDelivery = mailDelivery;
+        this.routes = simulation.getUniqueRoutes().stream().filter(this::isRouteSuitable).collect(Collectors.toList());
+
+    }
+
+    @SuppressWarnings("RedundantIfStatement")
+    private boolean isRouteSuitable(Route r) {
+        if(r.isDiscontinued()) {
+            return false;
+        }
+        if(r.getMaxVolume() < mailDelivery.getVolume() && r.getMaxVolume() != 0) {
+            return false;
+        }
+        if(r.getMaxWeight() < mailDelivery.getWeight() && r.getMaxWeight() != 0) {
+            return false;
+        }
+        return true;
     }
 
-    @Nullable public CalculatedRoute buildCalculatedRoute(@NotNull String from, @NotNull String to, @NotNull Priority priority) {
-        List<CalculatedRoute> possiblePaths = new ArrayList<>();
-        for (Route r : routes){
-            if (r.getFrom().equals(from)){
-                List<String> visitedNodes = new ArrayList<>();
+    @SuppressWarnings("Convert2streamapi") @Nullable
+    public CalculatedRoute buildCalculatedRoute() {
+        String from = mailDelivery.getFrom();
+        String goal = mailDelivery.getTo();
+
+        for (Route r : routes) {
+            if (r.getFrom().equals(mailDelivery.getFrom())){
                 visitedNodes.add(r.getFrom());
-                calculateCalculatedRoute(from, to, new CalculatedRoute(), possiblePaths, visitedNodes);
+                calculateRoute(from, goal, null);
             }
         }
-
         if(possiblePaths.size() == 0) {
             return null;
         }
-        return possiblePaths.stream().min((a, b) -> Integer.compare(getFit(a, priority), getFit(b, priority))).get();
-
+        return possiblePaths.stream().min((a, b) -> Integer.compare(getFit(a, mailDelivery.getPriority()), getFit(b, mailDelivery.getPriority()))).get();
     }
 
-    private void calculateCalculatedRoute(@NotNull String from, @NotNull String goal, @NotNull CalculatedRoute currentRoute, @NotNull List<CalculatedRoute> PossiblePaths, @NotNull List<String> visitedNodes) {
-        for (Route r : routes){
-            if (r.getFrom().equals(from)){
+    private void calculateRoute(@NotNull String from, @NotNull String goal, @Nullable CalculatedRoute currentRoute) {
+        if(currentRoute == null) {
+            currentRoute = new CalculatedRoute();
+        }
```

Comment:
I was thinking about potential bugs it could introduce (like resetting the calculated route), but I can't see that happening with the way the method has been written.